### DISCUSSION TITLE
Plotting Scalar density into IPF. Scalar interpolation in orientation space.

### DISCUSSION THREAD

**Discussion by fgallardo-mpie** [mtex-toolbox/mtex#1076] [Ask Anything] [answered]
Hello,

I want to plot the density function of a scalar in an IPF triangle. I am doing BCC (Cubic).
I have seen these done for schmid factor, taylor factor, etc.

The data I have is a 4 column array with the 3 euler angles and the scalar.
I first take into consideration the extra 23 orientations given one (ebsd.orientations.symmetrise). 
These 24 orientations correspond to the same point in the IPF triangle (if i am right)
I interpolate this scalar in the orientation space using interp.
Then I plot.

For clarity, i took my scalar to be (1,2,3,4,5...,25) starting from the top of the triangle. 
ebsd.id will be my scalar, but only for this example, the idea is to change it later on

```
sF   = interp( inv ( ebsd.orientations.symmetrise ) * vector3d.Z, repelem(ebsd.id,48), 'harmonicApproximation');
contourf(sF, [0:1:26],'ShowText','on','FontSize',15) %use 'contour' if you want no fill
mtexColorbar
hold on
plotIPDF(ebsd.orientations,ebsd.id,zvector,'MarkerSize',5)
mtexColorbar
colormap parula
%draw square where points are
for i=1:26
    hkl=ebsd.orientations(i).round2Miller
    annotate(ebsd.orientations(i),...
    'marker','s','MarkerSize',10,'MarkerFaceColor','none','color','k', 'markerEdgeColor', 'k')
drawNow(gcm,'figSize','normal')
end
mtexColorbar
```

![ipftriangle](https://user-images.githubusercontent.com/54398148/120815663-52138a80-c550-11eb-8480-8aac32a82f43.png)

Question 1: How can i get the labels of the levels as it happens for level '22'. Why do the others don't show?

Question 2: How can i improve the interpolation? I realize we are interpolation in the orientation space. Can i change the harmonic function parameters somehow in interp ???

Many thanks,
Fran

PS: I saw many questions about plotting scalars into an IPF. Use the code, but make sure what you plot make sense (symmetry, is scalar ipf dependant or orientation dependent, etc...)

**Comment by kilir** [comment, marked as answer]
Hi,
wrt 1) this is somewhat a bug since Mtex plots the contours slightly differently and hereby it seems to loose the labels somehow. You may have luck with using the `'LabelSpacing'` option. Funnily the labels show if you plot the complete hemisphere using `'complete'`.

wrt 2) In your example the interpolation is not in orientations space but on the sphere (you use directions as input for interp). For the harmonicApproximation method, you can use the option `'bandwidth'` to get control of the smoothness of the interpolation.

Hope that helps.

Cheers,
Rüdgier
